### PR TITLE
Comment and String locations

### PR DESCRIPTION
**Pull request by schloerke**
Hello,

The changes in this PR fix the issues in https://github.com/graphql/libgraphqlparser/issues/51 .  
 
* Lines were not tracked when comments were reached
* STRING_STATE routine did not set the `*yylloc` to `yyextra->loc`, similar to other tokens

Best,
Barret

### CODE REVIEW

File: lexer.cpp [hunk 1]
```diff
@@ -1,15 +1,15 @@
-#line 1 "lexer.cpp"
+#line 2 "/Users/barret/_/git/R/libgraphqlparser/libgraphqlparser/lexer.cpp"
```
Comment (swolchok) [inline lexer.cpp]: this needs to be fixed up. I think cmake is running flex incorrectly; I just run it manually for commit.

**Comment by schloerke** [comment]
@swolchok Cleaned up the files with the correct flex version.  (Much cleaner PR)

**Comment by swolchok** [comment]
Sorry this took me so long to get to! Had a baby in April & thusly have been super busy.

This looks mergeable to me. Worst case, the offsets are wrong in a different way, which doesn't leave us worse off. :)

Can you clean up the commit history? Up to you whether to squash it into a single commit or a couple commits, but we shouldn't end up with "undo whitespace", "remove personal paths", etc. in our history.

**Comment by schloerke** [comment]
Congratulations!

I have force updated the PR.  It is much cleaner now.

**Comment by swolchok** [comment]
Thanks, that's really easy to follow.

**Pull request merged**